### DISCUSSION TITLE
Error with sass-loader possibly: "Cannot find module 'sass'" when building to DEV environment

### DISCUSSION THREAD

**Discussion by wangytangy** [webpack/webpack#16663] [Q&A]
Hi Webpack community, I need help resolving a bug that I cannot solve on own. My team is trying to deploy our website after upgrading to Node v16 and Webpack v5. The problem is our webpack "sass-loader" is erroring out because it claims the "sass" module is not found.

_NOTE: our current config works on our localhost but not when deployed in a DEV environment!_

The error in our logs:
```
uncaughtException: Cannot find module 'sass'
Require stack:
- /var/node/atlas/build/webpack.config.js
- /var/node/atlas/server/main.js
- /var/node/atlas/bin/server.js

at Object.<anonymous> (/var/node/atlas/build/webpack.config.js:119:21)
```

Our webpack config file looks like this. Our setup is exactly what the Webpack documentation recommends:
```
//webpack.config.js
115 const sassLoader = {
116  loader: 'sass-loader',
117  options: {
118    sourceMap: true,
119    implementation: require('sass') /** THIS IS FAILING according to our logs */
120  }
121 };

// how sassLoader is being used
webpackConfig.module.rules.push({
  test: /\.scss$/,
  include: isInClientFolder,
  exclude: isGlobalStyle,
  use: [
    'style-loader',
    cssLoader,
    sassLoader
  ]
});
```

We checked to make sure our **package.json** has all libraries that need to be installed. It seems fine:
```
// package.json
"devDependencies": {
    "sass": "^1.57.1",
    "sass-loader": "^13.0.2",
    "style-loader": "^3.3.1",
    "webpack": "^5.74.0",
    "webpack-dev-middleware": "^5.3.3",
    "webpack-hot-middleware": "^2.25.2"
  }
```

Last of all, we tried installing Sass globally on our **Dockerfile** but still getting the error:
```
// Dockerfile
FROM node:16.19.0

RUN npm install npm@8.19.3 -g

RUN npm config set @dpb-packages:registry=https://dpb-repositories-183626424367.d.codeartifact.us-east-1.amazonaws.com/npm/npm-repository/ && \
    npm config set //dpb-repositories-183626424367.d.codeartifact.us-east-1.amazonaws.com/npm/npm-repository/:_authToken=$CODEARTIFACT_AUTH_TOKEN && \
    npm config set //dpb-repositories-183626424367.d.codeartifact.us-east-1.amazonaws.com/npm/npm-repository/:always-auth=true && \
    npm cache clear --force && \
    npm install && \
    npm install -g sass@1.57.1 && \
    mv node_modules /var/node/atlas
```

Currently this configuration does not work and we are still getting the _"Cannot find module 'sass'"_ error. 

I've already tried:
1. Moving 'sass' in package.json from devDependencies to dependencies but it breaks other loaders in Webpack.
2. I've tried installing Sass globally in Dockerfile but it didn't change anything
3. I've used the default Webpack sass-loader by just using a string 'sass-loader' in the webpack config but also broke the other loaders
4. Tried configuring the sass-loader options to turn off "fibers" which has to do with sync/async loading of the 'sass' package but it didn't do anything. And also Webpack docs say "fibers" doesn't work with Node v16 and up.

Can anyone offer any other ideas on how to fix this?

Thank you!

**Comment by alexander-akait** [comment]
Check your `NODE_PATH` in the docker container, I think something wrong here and Node.js can't find the `sass` package, also please try to run `npm ls sass` and `npm ls sass-loader` insode the docker container and provide output, thank you
